main.go

```diff
@@ -87,16 +88,71 @@ func myLANIP() (string, error) {
 	return "", err
 }
 
+func nmapScanAll() {
+	myIP, err := myLANIP()
+	handleErr(err)
+	out, err := exec.Command("nmap", "-A", "-T4", myIP).Output()
+	handleErr(err)
+	log.Println(string(out))
+}
 func nmapEntries() {
 	myIP, err := myLANIP()
 	handleErr(err)
-	out, err := exec.Command("nmap", "-sP", myIP).Output()
+	out, err := exec.Command("nmap", "-sP", "-PA21,22,25,3389", myIP).Output()
 	handleErr(err)
 	log.Println(string(out))
 }
 
+type Event struct {
+	When time.Time
+	Up   bool
+}
+
+func (e Event) String() string {
+	var s string
+	if e.Up {
+		s = "UP"
+	} else {
+		s = "DOWN"
+	}
+
+	return fmt.Sprintf("Event %s at %s", s, e.When)
+}
+func checkPing(h Host) Event {
+	out, err := exec.Command("nmap", "-sP", h.IP).Output()
+	if err == nil {
+		if strings.Contains(string(out), "1 host up") {
+			return Event{time.Now(), true}
+		}
+	}
+	return Event{time.Now(), false}
+}
+
+func checkAll(hosts []Host, ch chan Event, chost chan Host) {
```

Comment:
Why does this function needs another channel to store `Host` while its input is also a list of `Host`